File: 01/day1.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include <algorithm>

#include "../utils/utils.h"

using namespace std;

int main() {
  vector<string> lines = open_file("01/day1.txt");
  vector<int> entries;
  transform(lines.begin(), lines.end(), back_inserter(entries), [](const string s) {return stoi(s);});

  int answer = 0;
  for (int i = 1; i < entries.size(); ++i) {
    if (entries[i] > entries[i-1]) answer += 1;
  }

  cout << answer << endl;

  answer = 0;
  int sum1 = 0;
  int sum2 = 0;
  for (int i = 4; i < entries.size(); ++i) {
    sum1 = 0;
    sum2 = 0;
    for (int j = (i-3); j < i; ++j) {
      sum1 += entries[j];
      sum2 += entries[j-1];
    }

    if (sum1 > sum2) answer += 1;
  }

  cout << answer << endl;

  return answer;
}

File: python/day1.py
with open('01/day1.txt', 'r') as f:
  lines = f.readlines()

entries = [int(string.strip()) for string in lines]

answer = 0
for i in range(1, len(entries)):
  if entries[i] > entries[i-1]:
    answer += 1

print(answer)

answer = 0
for i in range(4, len(entries)):
  if sum(entries[i-3:i]) > sum(entries[i-4:i-1]):
    answer += 1

print(answer)